src/yaml-tools.js

```diff
@@ -111,11 +123,30 @@ yamlLibrary = {
   // doc: The SAM.YML parsed document
   // schema: the JSON schema to compare the doc to
   checkDocAgainstSchema: function (doc, schema, failed, warning, info) {
-    this.doc = doc;
+    this.doc = this.docs[doc];
     var keys = [];
 
     this.recurseIntoArray(schema, keys, null, failed, warning, info);
+  },
+
+  loadDocData: function (docs) {
+    let docsArray = cleanDocsData(docs).split('---');
+
+    for (let i = 0; i < docsArray.length; i++) {
+      let allComments = true;
```

Comment:
What is the purpose of allComments?  It doesn't seem like the code is actually doing anything with it.

Reply:
You're right, this was left over from before I added the cleaning script...this whole loop can be cleaned up.